### ISSUE TITLE
Use default argument names in lifecycle methods?

### ISSUE THREAD

**Issue by mike391** [yannickcr/eslint-plugin-react#2213]
Is it possible to enforce default arguments when using lifecycle methods?  For example:

```  componentDidUpdate(lastProps) {
}```

should complain and correct to this:

```  componentDidUpdate(prevProps) {
}```

**Comment by ljharb** [comment]
I suppose we could make a rule that did that; would that just ignore destructured arguments?

**Comment by mike391** [comment]
Yea, I was thinking it could be helpful for when one thinks that `componentDidUpdate` uses `nextProps` instead of `prevProps`, etc

**Comment by ljharb** [comment]
I feel like that might be better addressed with an editor autocomplete, rather than an eslint rule, though?
